static/app/views/ddm/createAlertModal.tsx

```diff
@@ -119,6 +111,35 @@ function getAlertInterval(metricsQuery, period: TimePeriod) {
   return toInterval(TimeWindow.ONE_HOUR);
 }
 
+export function getAlertFormPath({
+  organizationSlug,
+  interval,
+  statsPeriod,
+  environment,
+  project,
+  aggregate,
+  query,
+}) {
+  return `/organizations/${organizationSlug}/alerts/new/metric/?${qs.stringify({
+    // Needed, so alerts-create also collects environment via event view
+    createFromDiscover: true,
+    dataset: Dataset.GENERIC_METRICS,
+    eventTypes: EventTypes.TRANSACTION,
+    aggregate,
+    interval,
+    statsPeriod,
+    referrer: 'ddm',
+    // Event type also needs to be added to the query
+    query,
```

Comment:
Does it now work without? 🤔 
btw that comment disappeared in the other place where the url is constructed.
Maybe a common util would be nice so we keep the weirdness in one place?